support/checkLinks.js

```diff
@@ -49,11 +49,27 @@ async function checkInternal(link, files) {
 	for (const file of files) {
 		const contents = await readFileAsync(file, 'utf8');
 
-		const headingPattern = link.slice(1).replace(/-/g, '[ -,]+');
-
-		const regExp = new RegExp(`^#+\\s+${headingPattern}\\s*$`, 'im');
+		const targets = new Set();
+
+		contents.replace(/^#+\s+(.+?)\s*$/gm, (match, heading) => {
+			// To make a title anchor, GitHub:
+			// - Extracts link targets from Markdown links.
+			// - Turns spaces, hyphens, commas into hyphens.
+			// - Removes backticks, colons.
+			const target = heading
+				.replace(/^\[([^\\]+)\].*$/, (_match, text) => {
+					return text;
+				})
+				.replace(/[ -,]+/g, '-')
+				.replace(/[`:]/g, '');
+
+			// Lowercase because the browser matches case-insensitively.
+			targets.add(target.toLowerCase());
+
+			return match;
+		});
 
```

Comment:
I'm aware that this isn't super efficient. Choosing not to prematurely optimize here because the lack of efficiency isn't actually a problem (number of documents is so small), and that way we can keep things simple. Full check (including network access to check remote links, and `yarn` start-up time) takes 3.6s.